### PR TITLE
Update architecture.md

### PR DESCRIPTION
**Pull request by andrewgordstewart**
This is overdue, after @lalexgap's changes to the wallet API.

- add message service
- remove `scope` from objective description

### CODE REVIEW

File: packages/server-wallet/docs/architecture.md [hunk 1]
```diff
@@ -34,39 +34,56 @@ A state channel wallet has the following _runtime dependencies_:
...
+          +------^-----+                                               
+                 |                                                     
+                 |                                                     
+                 |                                                     
+                 |                                                     
+                 |                                                     
+       +---------v--------+                                            
+       |      ENGINE      |                                            
+       |                  |                                            
+       |   +-----------+  |
```
Comment (lalexgap) [inline packages/server-wallet/docs/architecture.md:76]: Currently both the `wallet` and `engine` access the `store`

**Review by lalexgap — APPROVED** [review]

File: packages/server-wallet/docs/architecture.md (hunk 1 above)
Comment (andrewgordstewart) [reply packages/server-wallet/docs/architecture.md:76]: What does the `wallet` use the store for?

Comment (lalexgap) [reply packages/server-wallet/docs/architecture.md:76]: The most common use case is updating chain data in the `store` from `ChainEvent`s from the chain service before cranking the engine.

Comment (andrewgordstewart) [reply packages/server-wallet/docs/architecture.md:76]: is that the _only_ use case?

In my head, I expected that the wallet would just push a chain event into the engine, just like it pushes a message into the engine. In that world, a wallet would not need a store.

Comment (andrewgordstewart) [reply packages/server-wallet/docs/architecture.md]: da76ba062b78d4f8c91fcd897507f0a1a43b19c0

**Pull request merged**